File: server/router/casework/casework.go
package casework

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type CaseWorkRouter struct{}

func (r *CaseWorkRouter) InitCaseWorkRouter(router *gin.RouterGroup) {
	writeGroup := router.Group("care").Use(middleware.OperationRecord())
	readGroup := router.Group("care")
	{
		writeGroup.POST("attention-cases/:id/acknowledge", caseWorkApi.AcknowledgeAttentionCase)
		writeGroup.POST("attention-cases/:id/handling-records", caseWorkApi.CreateHandlingRecord)
		writeGroup.POST("attention-cases/:id/escalate", caseWorkApi.EscalateAttentionCase)
		writeGroup.POST("attention-cases/:id/close", caseWorkApi.CloseAttentionCase)
		writeGroup.POST("attention-cases/:id/reopen", caseWorkApi.ReopenAttentionCase)
		writeGroup.POST("consultations/:id/assign", caseWorkApi.AssignConsultation)
		writeGroup.POST("consultations/:id/replies", caseWorkApi.ReplyConsultation)
		writeGroup.POST("consultations/:id/transfer", caseWorkApi.TransferConsultation)
		writeGroup.POST("consultations/:id/escalate", caseWorkApi.EscalateConsultation)
		writeGroup.POST("consultations/:id/resolve", caseWorkApi.ResolveConsultation)
		writeGroup.POST("consultations/:id/close", caseWorkApi.CloseConsultation)
		writeGroup.POST("consultations/:id/reopen", caseWorkApi.ReopenConsultation)
	}
	{
		readGroup.GET("workbench", caseWorkApi.GetWorkbench)
		readGroup.GET("attention-cases", caseWorkApi.ListAttentionCases)
		readGroup.GET("attention-cases/:id", caseWorkApi.GetAttentionCase)
		readGroup.GET("consultations", caseWorkApi.ListConsultations)
		readGroup.GET("consultations/:id", caseWorkApi.GetConsultation)
		readGroup.GET("consultations/:id/assignee-options", caseWorkApi.ListConsultationAssigneeOptions)
	}
}

File: web/src/api/sleep-care/case-work.js
import service from '@/utils/request'

const commandHeaders = (idempotencyKey) => ({
  'Idempotency-Key': idempotencyKey || crypto.randomUUID()
})

export const getCareWorkbench = () =>
  service({
    url: '/care/workbench',
    method: 'get'
  })

export const getAttentionCases = (params) =>
  service({
    url: '/care/attention-cases',
    method: 'get',
    params
  })

export const getAttentionCase = (id) =>
  service({
    url: `/care/attention-cases/${id}`,
    method: 'get'
  })

export const acknowledgeAttentionCase = (id, data, idempotencyKey) =>
  service({
    url: `/care/attention-cases/${id}/acknowledge`,
    method: 'post',
    data,
    headers: commandHeaders(idempotencyKey)
  })

export const createAttentionHandlingRecord = (id, data, idempotencyKey) =>
  service({
    url: `/care/attention-cases/${id}/handling-records`,
    method: 'post',
    data,
    headers: commandHeaders(idempotencyKey)
  })

export const escalateAttentionCase = (id, data, idempotencyKey) =>
  service({
    url: `/care/attention-cases/${id}/escalate`,
    method: 'post',
    data,
    headers: commandHeaders(idempotencyKey)
  })

export const closeAttentionCase = (id, data, idempotencyKey) =>
  service({
    url: `/care/attention-cases/${id}/close`,
    method: 'post',
    data,
    headers: commandHeaders(idempotencyKey)
  })

export const reopenAttentionCase = (id, data, idempotencyKey) =>
  service({
    url: `/care/attention-cases/${id}/reopen`,
    method: 'post',
    data,
    headers: commandHeaders(idempotencyKey)
  })
